pkg/apply/merge.go

```diff
@@ -10,11 +10,6 @@ import (
 // This is to be able to do a a meaningful comparison in apply,
 // since objects created on runtime do not have these fields populated.
 func MergeMetadataForUpdate(current, updated *uns.Unstructured) error {
-	updated.SetCreationTimestamp(current.GetCreationTimestamp())
```

Comment:
If all these are set to the same values as in `current` obj. how would it trigger the reconcile?

Reply:
If we keep using`DeepEqual `. There are many other object-specific fields that need to be set. , we need to create a very complicated logic to make sure that all necessary fields are copied before comparing.

Reply:
If we change DeepEqual to DeepDerivative,  do we still need to remove the update here?

Reply:
We can still leave the update here, but it would not make any difference. As if those fields are unset, they will not be token account of.

Reply:
My concern is If we remove the update, it may result in a CR without these fields.

Reply:
It will not happen, those metadata will be set by the server during the update.

Reply:
The values set by server will be different from previous (current) ones or they are the same?

Reply:
They will be the same, except for `generation`, it will increase by every change. Actually those metadata are managed by the server, and cannot be set by a client. I think the old code here is just for fulfilling the comparison requirement of `DeepEqual`.

Reply:
ok